pkg/client/cache/delta_fifo.go

```diff
@@ -99,7 +101,9 @@ type DeltaFIFO struct {
 
 	// knownObjectKeys list keys that are "known", for the
 	// purpose of figuring out which items have been deleted
-	// when Replace() is called.
+	// when Replace() is called. If the given KeyLister also
+	// satisfies the KeyGetter interface, the deleted object
+	// will be included in DeleteFinalStateUnknown markers.
```

Comment:
gotcha, my misunderstanding. Since you moved the important detail, can you put this comment back the way it was before your change? sorry for excessive nits here...